tests/vmi_configuration_test.go

```diff
@@ -237,6 +237,63 @@ var _ = Describe("[sig-compute]Configurations", func() {
 			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
 			libvmi.WithNetwork(v1.DefaultPodNetwork()),
 		)
+
+		DescribeTable("with memory configuration", func(guestMemory *int, limitsMemory *int, requestsMemory *int) {
+			vmi := libvmi.NewAlpine()
+			var expectedGuestMemory int
+
+			// Only Guest memory is provided
+			if guestMemory != nil {
+				guestMemReq := resource.MustParse(fmt.Sprintf("%dMi", *guestMemory))
+				vmi.Spec.Domain.Memory = &v1.Memory{Guest: &guestMemReq}
+				expectedGuestMemory = *guestMemory
+			}
+
+			vmi.Spec.Domain.Resources = v1.ResourceRequirements{}
+
+			// Only resources limits provided (same as limits == requests)
+			if limitsMemory != nil && requestsMemory == nil {
+				expectedGuestMemory = *limitsMemory
+				vmi.Spec.Domain.Resources = v1.ResourceRequirements{
+					Limits: kubev1.ResourceList{
+						kubev1.ResourceCPU:    resource.MustParse("1"),
+						kubev1.ResourceMemory: resource.MustParse(fmt.Sprintf("%dMi", *limitsMemory)),
+					},
+				}
+			}
+
+			// resources limits and requests are different
+			if limitsMemory != nil && requestsMemory != nil {
+				expectedGuestMemory = *requestsMemory
+				vmi.Spec.Domain.Resources = v1.ResourceRequirements{
+					Requests: kubev1.ResourceList{
+						kubev1.ResourceCPU:    resource.MustParse("1"),
+						kubev1.ResourceMemory: resource.MustParse(fmt.Sprintf("%dMi", *requestsMemory)),
+					},
+					Limits: kubev1.ResourceList{
+						kubev1.ResourceCPU:    resource.MustParse("1"),
+						kubev1.ResourceMemory: resource.MustParse(fmt.Sprintf("%dMi", *limitsMemory)),
+					},
+				}
+			}
+
+			By("Starting a VirtualMachineInstance")
+			vmi = tests.RunVMIAndExpectScheduling(vmi, 60)
+			tests.WaitForSuccessfulVMIStart(vmi)
+
+			expectedMemoryInKiB := expectedGuestMemory * 1024
+			expectedMemoryXMLStr := fmt.Sprintf("unit='KiB'>%d", expectedMemoryInKiB)
+
+			domXml, err := tests.GetRunningVirtualMachineInstanceDomainXML(virtClient, vmi)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(domXml).To(ContainSubstring(expectedMemoryXMLStr))
+
+		},
+			Entry("provided by domain spec directly", pointer.Int(512), nil, nil),
```

Comment:
This could be simpler by using options. `WithGuestMemory`, `WithLimitMemory` `WithResourceMemory`. You just need to use `New` rather than `NewAlpine` to not get the default.